apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -1103,29 +1103,8 @@ def _find_extent_for_domain(
     if np.issubdtype(dtype, NPInteger) or np.issubdtype(dtype, NPFloating):
         return min(extent, hi - lo + 1)
 
-    if dtype == "datetime64[s]":
-        ilo = int(lo.astype("int64"))
-        ihi = int(hi.astype("int64"))
-        iextent = min(extent, ihi - ilo + 1)
-        return np.datetime64(iextent, "s")
-
-    if dtype == "datetime64[ms]":
-        ilo = int(lo.astype("int64"))
-        ihi = int(hi.astype("int64"))
-        iextent = min(extent, ihi - ilo + 1)
-        return np.datetime64(iextent, "ms")
-
-    if dtype == "datetime64[us]":
-        ilo = int(lo.astype("int64"))
-        ihi = int(hi.astype("int64"))
-        iextent = min(extent, ihi - ilo + 1)
-        return np.datetime64(iextent, "us")
-
-    if dtype == "datetime64[ns]":
-        ilo = int(lo.astype("int64"))
-        ihi = int(hi.astype("int64"))
-        iextent = min(extent, ihi - ilo + 1)
-        return np.datetime64(iextent, "ns")
+    if dtype in ("datetime64[s]", "datetime64[ms]", "datetime64[us]", "datetime64[ns]"):
+        return min(extent, _util.to_unix_ts(hi) - _util.to_unix_ts(lo) + 1)
 
     return extent
```

Comment:
this function should not be typed to return `Any`.  Looking at the code paths, it should be typed  `int | Literal['']`

I realize this pre-dates the PR, but we should be fixing this stuff (using `Any` effectively turns off static type checking)

Reply:
This ended up being complicated to do in this PR so filed this: https://app.shortcut.com/tiledb-inc/story/65378/typing-for-domain-should-be-tuple-int-float-literal-int-float-literal